src/pages/Home/components/Portfolio/Prompt/Prompt.tsx

```diff
@@ -0,0 +1,508 @@
+            item.symbol === toTokenAddress ||
+            ('address' in item && item.address === toTokenAddress),
+        );
+        if (
+          !srcToken ||
+          (srcToken.type !== TokenType.ERC20 &&
+            srcToken.type !== TokenType.NATIVE)
+        ) {
+          throw new Error(`Cannot find the source token`);
+        }
+        if (
+          !toToken ||
+          (toToken.type !== TokenType.ERC20 &&
+            toToken.type !== TokenType.NATIVE)
+        ) {
+          throw new Error(`Cannot find the destination token`);
+        }
+
+        const amountBigInt = stringToBigint(
+          amount.toString(),
+          srcToken.decimals,
+        );
+
+        const rate = await getRate({
+          srcDecimals: srcToken.decimals,
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          destDecimals: toToken.decimals,
+          srcAmount: amountBigInt.toString(),
+        });
+        if (
+          isAPIError(rate.optimalRate) ||
+          !rate.optimalRate ||
+          !rate.destAmount
+        ) {
+          throw new Error(
+            `Error while getting swap rate. ${isAPIError(rate.optimalRate) ? rate.optimalRate.message : ''}`,
+          );
+        }
+
+        await swap({
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          srcDecimals: srcToken.decimals,
+          destDecimals: toToken.decimals,
+          srcAmount: rate.optimalRate.srcAmount,
+          priceRoute: rate.optimalRate,
+          destAmount: rate.destAmount,
+          slippage: 0.15,
+        });
+
+        return {
+          content: `Swap initiated ${amount}${srcToken.symbol} to ${rate.destAmount}${toToken.symbol}.`,
+        };
+      },
+    }),
+    [
+      accounts.active,
+      createContact,
+      getRate,
+      network,
+      request,
+      selectAccount,
+      swap,
+      t,
+      tokens,
+    ],
+  );
+
+  const systemPrompt = useMemo(() => {
+    if (!network || !tokens) {
+      return '';
+    }
+    return systemPromptTemplate
+      .replace(
+        '__TOKENS__',
+        JSON.stringify(
+          tokens.map((token) => ({
+            name: token.name,
+            symbol: token.symbol,
+            balance: token.balanceDisplayValue,
+          })),
+          (_, v) => (typeof v === 'bigint' ? v.toString() : v),
+        ),
+      )
+      .replace(
+        '__NETWORKS___',
+        JSON.stringify(
+          networks.map((n) => ({
+            id: n.caipId,
+            name: n.chainName,
+            isTestnet: n.isTestnet,
+          })),
+        ),
+      )
+      .replace(
+        '__CURRENT_NETWORK___',
```

Comment:
This does not get replaced
```suggestion
        '__CURRENT_NETWORK_ID__',
```